### PR TITLE
feat: UI to show authentication providers and logins

### LINKED ISSUE #1705: Adds a UI section in Podman Desktop to see all authentication providers / sign-in
### Is your enhancement related to a problem? Please describe

It will be possible to hook external authentication providers
- #1702 

And to signup for register
- #1704 

### Describe the solution you'd like

Authentication providers should be listed in the UI of Podman Desktop.
- Have a way to remove sessions (logout) for a given Authentication Provider

we should also have a way to Signup using a form

Let say we want to signup for 'FooBar' provider (if supported)

We collect all contributed properties with the `SignupProvider` scope
We display them in a nice way (if email type, display it as email with expected format, etc)

Once we have the data, we send the JSON with the name of the fields and their values to the `signup` method of a provider (if supported)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by dgolovin**
### What does this PR do?

First iteration implementation. The limitations are:
1. No Add account due to no support for multiple sessions
2. No hidden password due to not supporting password based authentication providers and short live time for SSO tokens
3. Login should be triggered only in context of client extension, so there so no login button for providers on this page
4. No Icon support in authentication API yet

### Screenshot/screencast of this PR

https://user-images.githubusercontent.com/620330/230538253-8a72ecbb-b341-4aae-80ea-90061ce40254.mov

It also supports keyboard navigation and execution for Log Out button.

https://user-images.githubusercontent.com/620330/230539223-af4c34a4-40c1-4dd4-87c7-163f2f25e0a3.mov

### What issues does this PR fix or reference?

Related to #1705.

### How to test this PR?

Build and install Red Hat SSO Authentication extension form https://github.com/redhat-developer/podman-desktop-redhat-account-ext and use menu to sign in to Red Hat SSO.

### CODE REVIEW

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte]:
```suggestion
              <!-- Authentication Contributor name -->
```

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte
Comment (deboer-tim) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte]: It may have been released after you started this PR, but please use SettingsPage component here (see other pages) to have a consistent page header.

**Review by jeffmaury — COMMENTED** [review]
This is not aligned with @mairin mockup for authentication is it planned for another issue ?

**Comment by dgolovin** [comment]
> This is not aligned with @mairin mockup for authentication is it planned for another issue ?

Implemented.

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte]: we should use `<button>` for all `on:click` calls so it's compliant with a11y

(it avoids also href="javascript:undefined")

File: packages/renderer/src/stores/authenticationProviders.ts
Comment (benoitf) [inline packages/renderer/src/stores/authenticationProviders.ts]:
```suggestion
export async function fetchAuthenticationProvidersInfo() {
```
small typo

File: packages/renderer/src/stores/authenticationProviders.ts
Comment (benoitf) [inline packages/renderer/src/stores/authenticationProviders.ts]:
```suggestion
  // here we have to rebuild authentication providers with sessions
```

File: packages/renderer/src/stores/authenticationProviders.ts
Comment (benoitf) [inline packages/renderer/src/stores/authenticationProviders.ts]:
```suggestion
window.events?.receive('authentication-provider-update', fetchAuthenticationProvidersInfo);
```

**Review by benoitf — COMMENTED** [review]
could you please add a component test for PreferencesAuthenticationProvidersRendering.svelte

- check what is displayed if we don't have any providers
- ensure we do the right call when we click on the provider

you'll probably need to add `aria-label="signout"` on the button that has an icon so it'll be easy to grab them

**Comment by benoitf** [comment]
If there is no auth provider we have:

image

so either:

1. we use the <EmptyScreen> component that allows to display something like
image

2. or if there is no auth providers, we hide in the menu the `Authentication` link in the navbar

**Comment by benoitf** [comment]
with Red Hat SSO extension I'm able to see my authentication
image

**Comment by benoitf** [comment]
as a side note, I noticed that there is some copy/paste from VS Code stuff inside the extension

image

I guess it should not display here 'VS Code'

**Comment by mairin** [comment]
Mockup idea for empty list

Authentication (Empty List) (1)

**Comment by dgolovin** [comment]
Implemented the 'No authentication providers' page.



> as a side note, I noticed that there is some copy/paste from VS Code stuff inside the extension
> 
> image
> 
> I guess it should not display here 'VS Code'

I wonder how you triggered this, because I cannot see this screen. My guess is one would see it when using SSO first time.

Extension is using vscode-redhat-account client id the same one used in vscode. We should get our own client id 'podman-desktop'.

**Comment by benoitf** [comment]
> I wonder how you triggered this, because I cannot see this screen. My guess is one would see it when using SSO first time.
> Extension is using vscode-redhat-account client id the same one used in vscode. We should get our own client id 'podman-desktop'.

if you go to https://sso.redhat.com/auth/realms/redhat-external/account/#/applications you can check Applications (if you delete VS Code it'll prompt you again)

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte]: I think we need the aria-label

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.spec.ts
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.spec.ts]:
```suggestion
  const signoutButton = screen.getByRole('button', { name: 'Sign out of Test Account' });
```

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.spec.ts
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.spec.ts]:
```suggestion
   const signoutButton = screen.getByRole('button', { name: 'Sign out of Test Account' });
```

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.spec.ts
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.spec.ts]: you should mostly use the getBy methods instead of the findBy methods

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte]: you need to use <EmptyScreen> component

like for example in ContainerDetailsLogs page

```

<EmptyScreen
  icon="{NoLogIcon}"
  title="No Log"
  message="Log output of {container.name}"
  hidden="{noLogs === false}"
  style="background-color: {getPanelDetailColor()}" />
```

**Review by benoitf — APPROVED** [review]
Approving modulo fixing before the merge:

- you need to use <EmptyScreen> for the empty screen (no providers)
- use aria-label / getByRole (to be much more a11y compliant) for defining / searching buttons

File: packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte
Comment (dgolovin) [reply packages/renderer/src/lib/preferences/PreferencesAuthenticationProvidersRendering.svelte]: I could not figure out why the icon was not aligned correctly, when using EmptyScreen component, that is why I implemented it this way. I figured it out this time.

**Pull request merged**
